mq/mq.go

```diff
@@ -35,6 +35,24 @@ type Client struct {
 
 type Option func(c *Client) error
 
+// handleChannelClose waits for notifications about closing amqpChan.
+// When amqpChan is closed we close conn.
+// On normal channel close err will be nil, on exception error will not be nil.
+// It is a blocking call.
+// TODO we should consider better error handling, e.g. recreate amqpChan instead of closing connection
+func handleChannelClose(amqpChan *amqp.Channel, conn *amqp.Connection) {
+	err := <-amqpChan.NotifyClose(make(chan *amqp.Error))
+	log.Info("received amqp channel close notification")
+
+	if err != nil {
+		log.Error(err)
+	}
+
+	if err := conn.Close(); err != nil {
```

Comment:
if connection is closed, what will happen to the consumers?

Reply:
If connection is already closed here err will not be nil and it will be logged. Additionally, this code should handle closed connection https://github.com/trustwallet/go-libs/blob/9e8f4b7c65afbc66d82f86ca3bb40c18aed64a04/mq/mq.go#L157